### DISCUSSION TITLE
Unable to join group from Signal Desktop if registering with signal-cli

### DISCUSSION THREAD

**Discussion by augustindo** [AsamK/signal-cli#429] [Q&A] [answered]
- [x] I have searched open and closed issues for duplicates

Hi there,
First thanks for your tool, it's pretty useful!
I'm not sure that it's relevant to create this issue here, sorry if not.

### Bug Description

After registering to Signal thanks to `signal-cli` under `Debian GNU/Linux 10 (buster)` (with a basic mobile phone, no smartphone) by running ` signal-cli -u USERNAME register` and ` signal-cli -u USERNAME verify CODE`, I'm unable to join groups with `Signal Desktop`.

Looking at [signalapp/Signal-Desktop/](https://github.com/signalapp/Signal-Desktop/) Github repo, I found this issue : [Failure to join v2 group from desktop client #4594 ](https://github.com/signalapp/Signal-Desktop/issues/4594) where @scottnonnenberg-signal says : 

> Seems that you're using Desktop standalone? It doesn't create a proper profile on account create like the mobile apps do, so you're not going to be able to join GroupV2 groups. You need a profileKey, and then a profileKeyCredential from the server, to join a GroupV2 group.

So I've created this feature request : [Allow to join groups in a Desktop standalone fashion #4850](https://github.com/signalapp/Signal-Desktop/issues/4850).

But **maybe something is doable on `signal-cli` side**.

### Steps to Reproduce

OS : `Debian GNU/Linux 10 (buster)`

1.  create a Signal account using  [signal-cli](https://github.com/AsamK/signal-cli) with SMS confirmation on basic mobile phone
2.  install Signal Desktop for GNU/Linux
3.  try to join a group on invitation by clicking on "Accept"
4. previous step fails with an error message. Logs says:

```
Top-level unhandled promise rejection: Error: promotePendingMember/groupv2([REDACTED]/eA=): No profileKeyCredential for conversation [REDACTED]2f7 ([REDACTED]3f7)
    at ConversationModel.promotePendingMember ([REDACTED]/app.asar/ts/models/conversations.js:221:23)
    at async [REDACTED]/app.asar/ts/models/conversations.js:292:37
```

Expected Result: be able to join groups.

### Platform Info

Signal-cli Version:
`0.7.2`

Signal Version:
`v1.39.5`

Operating System:
`Debian GNU/Linux 10 (buster) 64-bit`

**Comment by AsamK** [comment, marked as answer]
To be able to join a v2 group the account needs to have a profile (can be created with the `updateProfile` command)

**Reply by augustindo** [reply]
Thanks a lot for your answer ! Unfortunately I got the same error message.

### Steps to Reproduce

* run `./signal-cli -u +33600000000 updateProfile --name my_name --avatar ./my_avatar.jpg`
* restart `Signal Desktop`
* same error message when I try to accept an invitation. No `name` or `avatar` visible in `Signal Desktop`
* run `./signal-cli -u +33600000000 updateAccount`
* same problem (even with a new group which has been created and where I've been invited after last steps)

I understand that, in some way, it's out of `signal-cli` scope, but thanks in advance if you have any other advice, I've no other asset to resolve this problem.

**Reply by AsamK** [reply]
It seems Signal Desktop isn't aware of the new profile ... you could try to
- send a message from signal-cli to yourself `signal-cli -u +33600000000 send -m test +33600000000`, to transfer the profile key again
- if that doesn't work you could compile the latest master branch of signal-cli and try to update the profile again.
  I've added some code there to notify linked devices of the profile change (804ad34a0828d7b841f3fecd9af8f2bd948d31d3)
- or you could join the group with signal-cli, run `listGroups` to get the group id and join it with `updateGroup -g ID`

**Reply by augustindo** [reply]
Thanks a lot @AsamK, it worked !

I tried your first two steps but it did **not** work:

1. send a message to myself: it did not transfer my profile key
2. compile latest master branch and update the profile again

Then I tried your 3rd advice (`listGroups`) but result was empty. After running `signal-cli -u +33600000000 receive`, then `signal-cli -u +33600000000 listGroups` worked ! And I got my avatar up-to-date in `Signal Desktop`. Both joining groups with `signal-cli -u +33600000000 updateGroup -g ID` or through `Signal Desktop` GUI worked.

Thanks again for your help

**Reply by davidwitk** [reply]
Thanks, worked for me as well!

**Comment by mitar** [comment]
I got to the same problematic point. I first had to do `updateProfile` (without any arguments) which allowed others to start inviting me to v2 groups (without that they could not even invite me). But then I could not accept the invite in the Signal Desktop. `listGroups` did not work for me: it has not listed any group. But doing simply `sendContacts` allowed me then to accept the invite in Signal Desktop.

**Reply by friapro** [reply]
I tried everything in the [thread above](https://github.com/AsamK/signal-cli/discussions/429#discussioncomment-289229), but none of it worked. This did, however. Thank you!
